Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryAuditTransforms.java

```diff
@@ -129,19 +130,42 @@ private II getPatientIdFromRequest(PRPAIN201305UV02 request) {
             && request.getControlActProcess().getQueryByParameter().getValue() != null
             && request.getControlActProcess().getQueryByParameter().getValue().getParameterList() != null
             && request.getControlActProcess().getQueryByParameter().getValue().getParameterList()
-            .getLivingSubjectId() != null
-            && request.getControlActProcess().getQueryByParameter().getValue().getParameterList()
-            .getLivingSubjectId().get(0) != null
-            && request.getControlActProcess().getQueryByParameter().getValue().getParameterList()
-            .getLivingSubjectId().get(0).getValue() != null
-            && request.getControlActProcess().getQueryByParameter().getValue().getParameterList()
-            .getLivingSubjectId().get(0).getValue().get(0) != null) {
-
-            oII = request.getControlActProcess().getQueryByParameter().getValue().getParameterList().
-                getLivingSubjectId().get(0).getValue().get(0);
+            .getLivingSubjectId() != null) {
+
+            List<PRPAMT201306UV02LivingSubjectId> ids = request.getControlActProcess().getQueryByParameter().getValue()
+                .getParameterList().getLivingSubjectId();
+
+            if (ids.size() == 1) {
+                livingSubjectId = getLivingSubjectId(ids.get(0));
+            } else if (ids.size() > 1) {
+                // Get assignedDevice root
+                String root = request.getControlActProcess().getAuthorOrPerformer().get(0).getAssignedDevice()
```

Comment:
There are no null checks for getAuthorOrPerformer and all subsequent calls. Could throw null pointer or index out of bounds.  Are these checks done elsewhere?  Definitely would not want an exception thrown in auditing and cause message processing to fail.